src/guiFormSpecMenu.cpp

```diff
@@ -3580,6 +3536,20 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 			a->to_list = s.listname;
 			a->to_i = s.i;
 			m_invmgr->inventoryAction(a);
+
+			if (swap_items) {
+				if (button == MB_RIGHT)
```

Comment:
Note to reviewers: A little feature injection to set the new selected stack size depending on which button is pressed.